### PR TITLE
feat: add doc on setting up custom smtp provider

### PR DESCRIPTION
**Pull request by J0**
## What kind of change does this PR introduce?

see linked Notion doc

### CODE REVIEW

**Comment by GaryAustin1** [comment]
In looking at SES docs.
image
This mentions sending from EC2 instances.
If this applies to Supabase's gotrue service then it is a huge thing to mention when making a choice for early production products.

File: apps/docs/pages/guides/auth/auth-smtp.mdx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+export const meta = {
+  id: 'auth-smtp',
+  title: 'Configure a Custom SMTP',
+  description: 'Moving towards production: Configuring a custom SMTP provider',
+}
+
+## Overview
+
+At present, you can trial the Supabase platform by sending up to four emails per hour via the built service. As you progress toward production, you may find yourself wanting for a custom SMTP service in order to increase your limits. A custom SMTP server will allow you to set your own cap on the number of emails sent per hour.
```
Comment (darora) [inline apps/docs/pages/guides/auth/auth-smtp.mdx:11]: Suggest adding some language around the 4 emails/hour not being guaranteed, and the default email service as a whole being offered on a best-effort basis

File: apps/docs/pages/guides/auth/auth-smtp.mdx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+- Compliance and Anti Spam measures
+
+
+## How to Set up SMTP
+
+Head over to the [Auth Settings Page](https://supabase.com/dashboard/project/_/settings/auth) and hit "Enable Custom SMTP" under the SMTP Provider section.
+
+Fill in fields below with the relevant details obtained from your custom SMTP provider:
+
+[SMTP settings](/docs/img/guides/auth-smtp/smtp.png)
```
Comment (darora) [inline apps/docs/pages/guides/auth/auth-smtp.mdx:26]: Consider making the username in the screenshot something more like `user@example.com` (we should standardize on this across docs)

File: apps/docs/pages/guides/auth/auth-smtp.mdx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+## Configuring and Interpreting the email Rate limits
+
+The rate limit for sending emails is an aggregate number applied across all endpoints in the service which send emails. It also takes into account all users using the service. For example, if there are 50 users on the platform and the rate limit is `RATE_LIMIT`, the total number of emails that can be sent per hour is `RATE_LIMIT` and not `RATE_LIMIT` * 50.
+
+
+### SMTP Providers
+
+Some SMTP providers you could consider using are:
```
Comment (darora) [inline apps/docs/pages/guides/auth/auth-smtp.mdx:35]: Should make it clear that any SMTP provider is supported, and these are just some examples we've included

**Review by darora — APPROVED** [review]

**Review by gregnr — APPROVED** [review]

**Pull request merged**